dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheHttpClientBlockingChannel.java

```diff
@@ -70,7 +74,7 @@ public Response execute(Endpoint endpoint, Request request) throws IOException {
             RequestBody body = request.body().get();
             builder.setEntity(new RequestBodyEntity(body));
         }
-        return new HttpClientResponse(client.execute(builder.build()));
+        return responseLeakDetector.wrap(new HttpClientResponse(client.execute(builder.build())), endpoint);
```

Comment:
could we pull some of these out to variables so that it's not a one-liner? it makes stacktraces more immediately obvious and also makes stepping through in a debugger a little easier :)

Reply:
Updated, added another defensive check to close responses if exceptions are thrown between client.execute and returning a the dialogue response.